java/src/main/java/ai/rapids/cudf/ColumnVector.java

```diff
@@ -460,15 +460,52 @@ public static ColumnVector serial32BitMurmurHash3(int seed, ColumnView columns[]
   }
 
   /**
-   * Create a new vector containing the MD5 hash of each row in the table, seed defaulted to 0.
+   * Create a new vector containing the murmur3 hash of each row in the table, seed defaulted to 0.
    *
    * @param columns array of columns to hash, must have identical number of rows.
-   * @return the new ColumnVector of 32 character hex strings representing each row's hash value.
+   * @return the new ColumnVector of 32-bit values representing each row's hash value.
    */
   public static ColumnVector serial32BitMurmurHash3(ColumnView columns[]) {
     return serial32BitMurmurHash3(0, columns);
   }
 
+  /**
+   * Create a new vector containing spark's 32-bit murmur3 hash of each row in the table.
+   * Spark's murmur3 hash uses a different tail processing algorithm.
+   *
+   * @param seed integer seed for the murmur3 hash function
+   * @param columns array of columns to hash, must have identical number of rows.
+   * @return the new ColumnVector of 32-bit values representing each row's hash value.
+   */
+  public static ColumnVector spark32BitMurmurHash3(int seed, ColumnView columns[]) {
+    if (columns.length < 1) {
+      throw new IllegalArgumentException("MD5 hashing requires at least 1 column of input");
+    }
+    long[] columnViews = new long[columns.length];
+    long size = columns[0].getRowCount();
+
+    for(int i = 0; i < columns.length; i++) {
+      assert columns[i] != null : "Column vectors passed may not be null";
+      assert columns[i].getRowCount() == size : "Row count mismatch, all columns must be the same size";
+      assert !columns[i].getType().isDurationType() : "Unsupported column type Duration";
+      assert !columns[i].getType().isTimestampType() : "Unsupported column type Timestamp";
+      assert !columns[i].getType().isNestedType() : "Unsupported column of nested type";
+      columnViews[i] = columns[i].getNativeView();
+    }
```

Comment:
If the checks are the same for all of the hash functions should we make it common instead of copy/paste? Do you see us supporting different types for different hashes?

Reply:
Down the road I do see supporting different types and MD5 already supports 1-deep nested types. Adding nested types and timestamps is definitely on the roadmap for spark specific hashing but not necessarily for the serial murmur3. They are handled differently in each case, so the support isn't copyable between the two.